src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs

```diff
@@ -20,39 +19,50 @@ public override void ConfigureAppConfiguration(IFunctionsConfigurationBuilder bu
         {
             builder.ConfigurationBuilder.AddEnvironmentVariables();
         }
-
+        
         // This method gets called by the runtime. Use this method to add services to the container.
         // For more information on how to configure your application, visit https://docs.microsoft.com/en-us/azure/azure-functions/functions-dotnet-dependency-injection
         public override void Configure(IFunctionsHostBuilder builder)
         {
             IConfiguration config = builder.GetContext().Configuration;
-
+            
             builder.ConfigureSecretStore(stores =>
             {
 //[#if DEBUG]
                 stores.AddConfiguration(config);
 //[#endif]
-
+                
                 stores.AddEnvironmentVariables();
-
+                
                 //#error Please provide a valid secret provider, for example Azure Key Vault: https://security.arcus-azure.net/features/secret-store/provider/key-vault
                 stores.AddAzureKeyVaultWithManagedIdentity("https://your-keyvault.vault.azure.net/", CacheConfiguration.Default);
             });
-
-            var instrumentationKey = config.GetValue<string>("APPINSIGHTS_INSTRUMENTATIONKEY");
+            
+            LoggerConfiguration logConfig = CreateLoggerConfiguration(builder);
+            builder.Services.AddLogging(logging =>
+            {
+                logging.AddSerilog(logConfig.CreateLogger(), dispose: true);
+            });
+        }
+        
+        private static LoggerConfiguration CreateLoggerConfiguration(IFunctionsHostBuilder builder)
+        {
             var configuration = new LoggerConfiguration()
                 .MinimumLevel.Debug()
                 .MinimumLevel.Override("Microsoft", LogEventLevel.Information)
                 .Enrich.FromLogContext()
                 .Enrich.WithComponentName("Azure Databricks Metrics Scraper")
                 .Enrich.WithVersion()
-                .WriteTo.Console()
-                .WriteTo.AzureApplicationInsights(instrumentationKey);
-
-            builder.Services.AddLogging(logging =>
+                .WriteTo.Console();
+            
+            IConfiguration appConfig = builder.GetContext().Configuration;
+            var instrumentationKey = appConfig.GetValue<string>("APPINSIGHTS_INSTRUMENTATIONKEY");
+            if (instrumentationKey != null)
```

Comment:
Why not use String.IsNullOrWhitespace or something like that ?

Reply:
Will do.
Is still WIP (draft PR + 🚧).

Reply:
Done in https://github.com/arcus-azure/arcus.templates/pull/587/commits/844d1b8d5f71c36dc7c34c097bf47629a2088bd1